proctor-common/src/main/java/com/indeed/proctor/common/model/TestType.java

```diff
@@ -21,10 +23,13 @@ public final class TestType implements JsonSerializable {
     private static final ConcurrentMap<String, TestType> TYPES = Maps.newConcurrentMap();
     @Nonnull
     private final String name;
-
+    @Nonnull
+    private final Set<String> allowedDependencies;
```

Comment:
nits: I think `private final Set<String> allowedDependencies = new HashSet<>();` is a bit more clear than putting assignment into a different place (constructor).